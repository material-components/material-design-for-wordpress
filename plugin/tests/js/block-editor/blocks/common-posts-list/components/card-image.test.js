/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import CardImage from '../../../../../../assets/src/block-editor/blocks/common-posts-list/components/card-image';

/**
 * Render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <CardImage { ...props } /> );
};

const baseProps = {
	imageSourceUrl: 'http://example.com/test.jpg',
	type: '16-9',
	contentLayout: 'text-above-media',
};

describe( 'CardImage', () => {
	it( 'matches snapshot when the content layout is not "Text over media"', () => {
		const wrapper = setup( baseProps );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when the content layout is "Text over media"', () => {
		const props = {
			titleTrimmed: 'Test Title',
			displayPostDate: true,
			post: {
				date_gmt: ' 2020-02-25 08:20:38',
			},
			dateFormat: 'F j, Y',
			...baseProps,
		};

		props.contentLayout = 'text-over-media';

		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
