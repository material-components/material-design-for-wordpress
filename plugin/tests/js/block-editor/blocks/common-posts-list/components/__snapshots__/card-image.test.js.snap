// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImage matches snapshot when the content layout is "Text over media" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-card__media mdc-card__media--16-9 single-post-card__media single-post-card-with-text-over-media"
        style="background-image: url(http://example.com/test.jpg);"
      >
        <div
          class="mdc-card__media-content"
        >
          <div
            class="single-post-card__primary"
          >
            <h2
              class="single-post-card__title mdc-typography mdc-typography--headline6"
            >
              <div>
                Test Title
              </div>
            </h2>
            <h3
              class="single-post-card__subtitle mdc-typography mdc-typography--subtitle2"
            >
              <time
                datetime="2020-02-25 08:20:38"
              >
                February 25, 2020
              </time>
            </h3>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card__media mdc-card__media--16-9 single-post-card__media single-post-card-with-text-over-media"
      style="background-image: url(http://example.com/test.jpg);"
    >
      <div
        class="mdc-card__media-content"
      >
        <div
          class="single-post-card__primary"
        >
          <h2
            class="single-post-card__title mdc-typography mdc-typography--headline6"
          >
            <div>
              Test Title
            </div>
          </h2>
          <h3
            class="single-post-card__subtitle mdc-typography mdc-typography--subtitle2"
          >
            <time
              datetime="2020-02-25 08:20:38"
            >
              February 25, 2020
            </time>
          </h3>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CardImage matches snapshot when the content layout is not "Text over media" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-card__media mdc-card__media--16-9 single-post-card__media single-post-card-with-text-above-media"
        style="background-image: url(http://example.com/test.jpg);"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card__media mdc-card__media--16-9 single-post-card__media single-post-card-with-text-above-media"
      style="background-image: url(http://example.com/test.jpg);"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
